sycl/include/CL/sycl/detail/stream_impl.hpp

```diff
@@ -322,15 +323,54 @@ inline unsigned append(char *Dst, const char *Src) {
   return Len;
 }
 
+template <typename T>
+inline typename std::enable_if<std::is_same<T, half>::value, unsigned>::type
+checkForInfNan(char *Buf, T Val) {
+  if (Val != Val)
+    return append(Buf, "nan");
+
+  // Extract the sign from the bits
+  const uint16_t Sign = reinterpret_cast<uint16_t &>(Val) & 0x8000;
+  // Extract the exponent from the bits
+  const uint16_t Exp16 = (reinterpret_cast<uint16_t &>(Val) & 0x7c00) >> 10;
+
+  if (Exp16 == 0x1f) {
+    if (Sign)
+      return append(Buf, "-inf");
+    else
```

Comment:
Please, do not put `else` after `return`.

Reply:
Thx, fixed.